### PR TITLE
Visualization updates to support no results page

### PR DESCRIPTION
**Pull request by sarveshnagpal**
- Adding support for time to session visualization
- Explicitly handling when no results are found
- Minor visualization updates

### CODE REVIEW

File: src/visualization/components/Content.tsx [hunk 1]
```diff
@@ -6,7 +6,14 @@ import LinearProgress from 'material-ui/LinearProgress';
 class Content extends React.Component<any, any> {
 
     render() {
-        if (!this.props.impression) {
+        if (this.props.notfound) {
+            return (
+                <div className={'clarity-notfound'}>
+                    No match found.
```
Comment (toby-walker) [inline src/visualization/components/Content.tsx]: Very minor; Should what was searched and not found in the message here.

File: src/visualization/reducers/notfound.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { Types } from "../actions";
+
+export default function (state = false, action) {
```
Comment (toby-walker) [inline src/visualization/reducers/notfound.ts]: Minor: Add types for arguments.

File: src/extension/app.tsx [hunk 3]
```diff
@@ -41,9 +41,11 @@ chrome.runtime.sendMessage({ fetch: true }, function (response) {
             size += entry.length;
             let json = JSON.parse(uncompress(entry.payload));
             let tabId = entry.tabId;
+            let dateTime = entry.dateTime;
```
Comment (ender336) [inline src/extension/app.tsx]: >let dateTime = entry.dateTime; [](start = 12, length = 30)

nit: this seems a little more roundabount than doing 
json.envelope.dateTime = entry.dateTime then getting rid of line 48

File: src/visualization/reducers/impression.ts [hunk 4]
```diff
@@ -3,6 +3,11 @@ import { Types } from "../actions";
 export default function (state = null, action) {
     switch (action.type) {
         case Types.SelectSession:
+            for (var payload of action.payload) {
```
Comment (ender336) [inline src/visualization/reducers/impression.ts]: >var [](start = 17, length = 3)

nit: let

File: src/visualization/reducers/snapshot.ts [hunk 5]
```diff
@@ -3,12 +3,17 @@ import { Types } from "../actions";
 export default function (state = null, action) {
     switch (action.type) {
         case Types.SelectSession:
-            var payload =  action.payload[0];
-            var lastEvent = payload.events.length - 1;
-            return parseInt(payload.events[lastEvent].time);
+            for (var payload of action.payload) {
```
Comment (ender336) [inline src/visualization/reducers/snapshot.ts]: >var [](start = 17, length = 3)

nit: let

File: src/visualization/reducers/impression.ts (hunk 4 above)
Comment (ender336) [inline src/visualization/reducers/impression.ts]: >payload [](start = 39, length = 7)

also a nit, but payload is being used enough that I think this code has become confusing. Action can either have a payload in it, or a payload of payloads? Can we disambiguate the names slightly?

**Pull request merged**